guacamole/src/main/frontend/src/app/history/services/guacHistory.js

```diff
@@ -42,6 +42,31 @@ angular.module('history').factory('guacHistory', ['$injector',
      */
     service.recentConnections = [];
 
+    /**
+     * Remove from the list of connection history the item having the given
+     * identfier.
+     * 
+     * @param {String} id
+     *     The identifier of the item to remove from the history list.
+     *     
+     * @returns {boolean}
+     *     True if the removal was successful, otherwise false.
+     */
+    service.removeEntry = function removeEntry(id) {
+    
+        var i;
+        
+        for (i = 0; i < service.recentConnections.length; i++) {
+            if (service.recentConnections[i].id === id) {
```

Comment:
This definitely works fine. That said, we do have access to lodash in the webapp, which makes these sorts of operations a bit less clunky. 

For example, the contents of this function could be replaced by 

```
return _.remove(service.recentConnections, entry => entry.id === id).length > 0;
```

Reply:
Ah, cool, I'm happy to give it a shot with that.

Reply:
Thanks for the hint - this seems to work well!